research/error-correction/test_ecc.py

```diff
@@ -0,0 +1,42 @@
+import random
+import hamming
+
+
+def test_hamming_random_no_changes(tests=1000):
+    for i in range(tests):
+        size = random.randint(1, 2**10)
+        payload = [random.randint(0, 1) for _ in range(size)]
+        data = hamming.hamming_encode(payload)
+        assert hamming.hamming_decode(data) == payload
+
+
+def test_hamming_random_one_change(tests=1000):
+    for i in range(tests):
+        size = random.randint(1, 2**10)
+        payload = [random.randint(0, 1) for _ in range(size)]
+        data = hamming.hamming_encode(payload)
+
+        # Flip a random bit
+        n = random.randint(0, len(data)-1)
+        data[n] = 0 if data[n] == 1 else 1
+
+        assert hamming.hamming_decode(data) == payload
+
+def test_hamming_random_two_changes(tests=1000):
```

Comment:
Surely testing for a failure is an antipattern

Reply:
Fixed in [1617f3e](https://github.com/SuperZooper3/AVisualDataFormat/pull/6/commits/1617f3eaed09e130a86d3935285a09399704a8da)